kampkelly/MyDiaryApi
# Pt159268253 Feedback- Correctly implement api endpoints and error codes

#### What does this PR do?
This pull request addresses several issues on the endpoints that needed to be fixed to make the application work better.

#### Description of Task to be completed?
The status codes have now been properly edited, replacing error 500 with 400 where necessary.
Testing for response as an object has been removed.
The package.json has been tidied up.
Emails and password have now been stripped of spaces and converted to lowercase.
Have the following endpoints working:
Get api/v1/entries - to show all entries
Get api/v1/entries/:id - to show a single entry
Post api/v1/entries - to create an entry
Put api/v1/entries/:id - to update an  entry
Post api/v1/users -  to sign up
Post api/v1/users/signin -  to sign in
Put api/v1/users/:id -  to update a user profile
Get api/v1/users/:id -  to view a user profile
Put api/v1/:id/notifications - To set a user’s notifications

#### How should this be manually tested?
After cloning the repo, run ‘npm install’. Then run ‘npm run start’ to start the express server.
Then each of the endpoints can be tested using postman.
To test signup. The fields needed are fullName, email, password, confirm password and dob.
To signing, the fields are email and password.
To create an entry, the fields are title and description.

#### Any background context you want to provide?
This api is not using a persistent database 

#### What are the relevant pivotal tracker stories?
#159268253
#### Screenshots (if appropriate)
No screenshots
#### Questions:
